### PR TITLE
Remove dependency on profile and use local state "ads enabled" pref in stats updater

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves brave/brave-browser#26864
Resolves brave/brave-browser#4599

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Ensure the ads enabled parameter in usage ping reflects the ads enabled setting of the last loaded profile.

### CODE REVIEW

File: browser/brave_stats/brave_stats_updater_unittest.cc [hunk 1]
```diff
@@ -646,90 +623,97 @@ TEST_F(BraveStatsUpdaterTest, UsageBitstringNeverUsed) {
...
-  params->SavePrefs();
-
-  task_environment_.AdvanceClock(base::Days(1));
-  GetProfilePrefs()->SetTime(kBraveWalletLastUnlockTime, base::Time::Now());
-  params = BuildUpdaterParams();
-  url = params->GetUpdateURL(base_url, "", "", "");
-  EXPECT_THAT(url.query(), HasSubstr("daily=true&weekly=false&monthly=false"));
-  EXPECT_THAT(url.query(), HasSubstr("wallet2=1"));
-  params->SavePrefs();
+  // auto params = BuildUpdaterParams();
```
Comment (DJAndries) [inline browser/brave_stats/brave_stats_updater_unittest.cc:633]: will rebase and correct once #15952 is merged

File: browser/brave_stats/brave_stats_updater_params.cc [hunk 2]
```diff
@@ -120,8 +116,7 @@ void BraveStatsUpdaterParams::LoadPrefs() {
   last_check_month_ = stats_pref_service_->GetInteger(kLastCheckMonth);
   first_check_made_ = stats_pref_service_->GetBoolean(kFirstCheckMade);
   week_of_installation_ = stats_pref_service_->GetString(kWeekOfInstallation);
-  wallet_last_unlocked_ =
-      profile_pref_service_->GetTime(kBraveWalletLastUnlockTime);
+  wallet_last_unlocked_ = base::Time();
```
Comment (DJAndries) [inline browser/brave_stats/brave_stats_updater_params.cc:119]: will rebase and correct once #15952 is merged

File: components/brave_referrals/browser/brave_referrals_service.cc [hunk 3]
```diff
@@ -1,7 +1,7 @@
-/* Copyright 2019 The Brave Authors. All rights reserved.
```
Comment (simonhong) [inline components/brave_referrals/browser/brave_referrals_service.cc:1]: You can add `(c)` only to prevent presubmit error. We don't want to udpate year.

**Review by simonhong — APPROVED** [review]
++ `brave_referrals_service.cc`

File: browser/brave_ads/ads_stats_updater_helper.cc [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+#include "brave/components/brave_ads/common/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/common/pref_names.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/prefs/pref_service.h"
+
+namespace brave_ads {
+
+AdsStatsUpdaterHelper::AdsStatsUpdaterHelper()
```
Comment (tmancey) [inline browser/brave_ads/ads_stats_updater_helper.cc:18]: We don't need to prepend Ads because we are in the brave_ads namespace, I.e we could just have brave_ads::StatsUodaterHelper

**Review by tmancey — APPROVED** [review]
LGTM++

File: browser/brave_ads/ads_stats_updater_helper.cc (hunk 4 above)
Comment (DJAndries) [reply browser/brave_ads/ads_stats_updater_helper.cc:18]: I see. Was just following the convention of other files in that dir. will rename

Comment (tmancey) [reply browser/brave_ads/ads_stats_updater_helper.cc:18]: ah yes we are gradually removing the verbosity where it is not needed

Comment (DJAndries) [reply browser/brave_ads/ads_stats_updater_helper.cc:18]: ok noted, will keep this in mind going forward

File: browser/brave_browser_process_impl.cc [hunk 5]
```diff
@@ -423,3 +426,11 @@ brave::BraveFarblingService* BraveBrowserProcessImpl::brave_farbling_service() {
     brave_farbling_service_ = std::make_unique<brave::BraveFarblingService>();
   return brave_farbling_service_.get();
 }
+
+brave_ads::StatsUpdaterHelper*
+BraveBrowserProcessImpl::ads_stats_updater_helper() {
```
Comment (bridiver) [inline browser/brave_browser_process_impl.cc:431]: we don't need an accessor for this

File: browser/brave_ads/stats_updater_helper.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (bridiver) [inline browser/brave_ads/stats_updater_helper.h:1]: brave_stats_updater_helper

**Comment by bridiver** [comment]
need tests for stats_updater_helper

File: browser/brave_ads/ads_stats_updater_helper.cc (hunk 4 above)
Comment (bridiver) [reply browser/brave_ads/ads_stats_updater_helper.cc]: as mentioned in DM we do however need StatsUpdater -> BraveStatsUpdater to be consistent with other references in the code

File: test/base/testing_brave_browser_process.cc [hunk 7]
```diff
@@ -1,7 +1,7 @@
-/* Copyright (c) 2021 The Brave Software Team. Distributed under the MPL2
- * license. This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
```
Comment (bridiver) [inline test/base/testing_brave_browser_process.cc:1]: since there are no other changes please revert these back to the original so we don't have unrelated commits here

**Review by bridiver — APPROVED** [review]

File: test/BUILD.gn [hunk 8]
```diff
@@ -676,6 +681,7 @@ test("brave_browser_tests") {
     "//brave/app/brave_main_delegate_browsertest.cc",
     "//brave/app/brave_main_delegate_runtime_flags_browsertest.cc",
     "//brave/browser/brave_ads/ads_service_browsertest.cc",
+    "//brave/browser/brave_ads/brave_stats_updater_helper_browsertest.cc",
```
Comment (bridiver) [inline test/BUILD.gn:684]: this shouldn't be added directly here, it should go in a test target in the BUILD.gn for brave_stats and be added here (and for android browser tests) as a dep

**Review by bridiver — APPROVED** [review]

**Review by spylogsster — APPROVED** [review]

File: browser/brave_ads/brave_stats_updater_helper.cc [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+void BraveStatsUpdaterHelper::OnProfileAdded(Profile* profile) {
+#if BUILDFLAG(IS_ANDROID)
+  if (profile == ProfileManager::GetPrimaryUserProfile()) {
+#else
+  base::FilePath last_used_path =
+      local_state_->GetFilePath(::prefs::kProfileLastUsed);
+  if ((!last_used_path.empty() && profile->GetBaseName() == last_used_path) ||
+      (last_used_path.empty() &&
+       profile == ProfileManager::GetPrimaryUserProfile())) {
```
Comment (emerick) [inline browser/brave_ads/brave_stats_updater_helper.cc:50]: We shouldn't call `ProfileManager::GetPrimaryUserProfile`, as it's been deprecated upstream and removed in cr110:

```
  // WARNING: do not use this function on Desktop platforms (Windows, Mac,
  // Linux). See https://crbug.com/1264436 for more info.
  // TODO(https://crbug.com/1264436): restrict this function to Android and
  // ChromeOS.
  //
  // Get the profile for the user which created the current session.
  // Note that in case of a guest account this will return a 'suitable' profile.
  static Profile* GetPrimaryUserProfile();
```

We should probably address this in master and then I can pull the update into cr110 when rebasing next week, but happy to take a different approach too if it makes sense. Could we chat on Monday? cc: @tmancey

Comment (DJAndries) [reply browser/brave_ads/brave_stats_updater_helper.cc:50]: addressed via #16815
